.github/workflows/dists.yml

```diff
@@ -14,58 +14,82 @@ concurrency:  # https://stackoverflow.com/questions/66335225#comment133398800_72
+        - { os: macos-15,     target: x86_64,  manylinux: auto}
+        - { os: macos-15,     target: aarch64, manylinux: auto}
+        - { os: ubuntu-24.04, target: x86_64,  manylinux: auto}
+        - { os: ubuntu-24.04, target: aarch64, manylinux: auto}
+        - { os: ubuntu-24.04, target: i686,    manylinux: auto}
+        - { os: ubuntu-24.04, target: armv7,   manylinux: auto}
+        - { os: ubuntu-24.04, target: ppc64le, manylinux: auto}
+        - { os: ubuntu-24.04, target: s390x,   manylinux: auto}
+        - { os: ubuntu-24.04, target: x86_64,  manylinux: musllinux_1_1 }
+        - { os: ubuntu-24.04, target: aarch64, manylinux: musllinux_1_1 }
+        - { os: windows-2025, target: x86_64,  manylinux: auto}
+        - { os: windows-2025, target: i686,    manylinux: auto}
 
     runs-on: ${{ matrix.os }}
     steps:
     - uses: actions/checkout@v4
 
+    # keep python versions in sync with tests.yml
+    # all python versions need to be present for linking to succeed
     - uses: actions/setup-python@v5
       with:
-        # pin to the latest supported python version
-        python-version: "3.13"
-        # x86 python needs to be available for the win32 wheel
-        architecture: ${{ ( matrix.os == 'windows-latest' && matrix.target == 'i686' ) && 'x86' || null }}
+        python-version: '3.8'
+        architecture: ${{ ( startsWith(matrix.os, 'windows') && matrix.target == 'i686' ) && 'x86' || null }}
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.9'
+        architecture: ${{ ( startsWith(matrix.os, 'windows') && matrix.target == 'i686' ) && 'x86' || null }}
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.10'
+        architecture: ${{ ( startsWith(matrix.os, 'windows') && matrix.target == 'i686' ) && 'x86' || null }}
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.11'
+        architecture: ${{ ( startsWith(matrix.os, 'windows') && matrix.target == 'i686' ) && 'x86' || null }}
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+        architecture: ${{ ( startsWith(matrix.os, 'windows') && matrix.target == 'i686' ) && 'x86' || null }}
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.13'
+        architecture: ${{ ( startsWith(matrix.os, 'windows') && matrix.target == 'i686' ) && 'x86' || null }}
+    - uses: Quansight-Labs/setup-python@v5 # 3.13t support not yet merged ref https://github.com/actions/setup-python/pull/973#issuecomment-2495900996
+      with:
+        python-version: '3.13t'
+        architecture: ${{ ( startsWith(matrix.os, 'windows') && matrix.target == 'i686' ) && 'x86' || null }}
 
-    - uses: PyO3/maturin-action@v1
+    - uses: PyO3/maturin-action@v1.45.0
       with:
         target: ${{ matrix.target }}
-        manylinux: ${{ matrix.manylinux || 'auto' }}
-        # Keep in sync with tests.yml
+        manylinux: ${{ matrix.manylinux }}
```

Comment:
Just so I follow, what's the reason for this change?

Reply:
Ah I was experimenting with furthet breaking up the matrix, one Python version at a time. To keep that simple, I wrote out the full matrix, including the `auto` string. Ended up reverting, but decided to keep this one.